smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HttpBindingProtocolGenerator.kt

```diff
@@ -414,6 +416,31 @@ abstract class HttpBindingProtocolGenerator : ProtocolGenerator {
         }
     }
 
+    private fun renderUrlPathMiddleware(ctx: ProtocolGenerator.GenerationContext, op: OperationShape) {
+        val opIndex = OperationIndex.of(ctx.model)
+        val httpBindingResolver = getProtocolHttpBindingResolver(ctx, defaultContentType)
+        val httpTrait = httpBindingResolver.httpTrait(op)
+        val requestBindings = httpBindingResolver.requestBindings(op)
+        val pathBindings = requestBindings.filter { it.location == HttpBinding.Location.LABEL }
+        val inputShape = opIndex.getInput(op).get()
+        val outputShape = opIndex.getOutput(op).get()
```

Comment:
Seems like we should be using ServiceGenerator conveinence methods instead of operationIndex?

Reply:
Ah, i noticed that this might not be possible because we we are passing inputSymbol into HttpUrlPathMiddleware.  if this is the case, feel free to ignore this comment